app/api/ada/hardwareWallet/newTransaction.js

```diff
@@ -154,3 +159,90 @@ function _outputAddressOrPath(out: TxOutput) {
   }
   return { address: out.address };
 }
+
+/** Generate a payload for Ledger SignTx */
+export async function createLedgerSignTxPayload(
+  txExt: UnsignedTransactionExt
+): Promise<LedgerSignTxPayload> {
+
+  // Transactions Hash
+  const txDataHexList = await txsBodiesForInputs(txExt.inputs);
+
+  // Inputs
+  const ledgerInputs: Array<LedgerInputTypeUTxO> =
+    _transformToLedgerInputs(txExt.inputs, txDataHexList);
+
+  // Outputs
+  const ledgerOutputs = _transformToLedgerOutputs(txExt.outputs);
+
+  return {
+    inputs: ledgerInputs,
+    outputs: ledgerOutputs,
+  };
+}
+
+function _transformToLedgerInputs(
+  inputs: Array<TxInput>, 
+  txDataHexList: Array<string>
+): Array<LedgerInputTypeUTxO> {
+  return inputs.map((input: TxInput, idx: number) => ({
+    txDataHex: txDataHexList[idx],
+    outputIndex: input.ptr.index,
+    path: _derivePath(input.addressing.change, input.addressing.index),
+  }));
+}
+
+function _transformToLedgerOutputs(
+  outputs: Array<TxOutput>
+): Array<LedgerOutputTypeAddress | LedgerOutputTypeChange> {
+  return outputs.map(x => ({
+    amountStr: x.value.toString(),
+    ..._ledgerOutputAddressOrPath(x)
+  }));
+}
+
+function _ledgerOutputAddressOrPath(out: TxOutput) {
+  if (out.isChange) {
+    // It's a change output
+    const fullAddress: ?AdaAddress = out.fullAddress;
+    if (fullAddress) {
+      return { path: _derivePath(fullAddress.change, fullAddress.index) };
+    }
+
+    Logger.debug('ledgerNewTransactions::_ledgerOutputAddressOrPath: ' +
+      `[WEIRD] Trezor got a change output without a full 'Ada Address': ${stringifyData(out)}`);
+  }
+
+  return { address58: out.address };
+}
+
+/** Send a transaction and save the new change address */
+export async function newLedgerTransaction(
```

Comment:
@sarveshwar-singh this should be almost the same as Trezor, we shouldn’t create a new func

Reply:
have plan to delete this function after checking.

Reply:
please add TODOs

Reply:
@nicarq 
I had already add it before but don't know why it's not visible in Github.
Although will add more details to the TODOs.

Reply:
aha gotcha